core/src/main/java/org/apache/calcite/rel/RelDistributions.java

```diff
@@ -135,10 +140,16 @@ public RelDistribution apply(Mappings.TargetMapping mapping) {
       if (keys.isEmpty()) {
         return this;
       }
-      return getTraitDef().canonize(
-          new RelDistributionImpl(type,
-              ImmutableIntList.copyOf(
-                  Mappings.apply2((Mapping) mapping, keys))));
+      List<Integer> mapList = Mappings.asList(mapping);
+      for (Integer key : keys) {
```

Comment:
Is `int` OK here?

Reply:
I replaced `Mappings.asList(mapping)` with `mapping.getTargetOpt(key)`, please take a look.